packages/devtools/src/utils.ts

```diff
@@ -23,3 +23,15 @@ export function getRulesBySlots(node: DebugResult, result: SlotInfo[] = []): Slo
     return [...acc, ...getRulesBySlots(child)];
   }, result);
 }
+
+export function filterSlots(slots: SlotInfo[], searchTerm: string) {
+  if (!searchTerm.length) {
+    return slots;
+  }
+
+  const clone = JSON.parse(JSON.stringify(slots)) as SlotInfo[];
```

Comment:
the JSON functions are some of the most expensive ones in the browser, why do you need to deep clone the slot info ?

Reply:
heyy, the `slots` should not be changed because it contains all style information for the selected component. Search filters the slots and returns a subset. If I don't use a deep clone, changes will happen directly on 'slots', and when user clears filter I would have to reconstruct 'slots'.
I'm not happy with this implementation TBH, open for better suggestion.

Reply:
Just when I write reply to you I though about creating a new array as filtered result, instead of filtering on `slots` array. It is much faster. For 1000 simple search, JSON approach was 6.66ms and this one was 2.91ms

Reply:
yeah, it is never better to serialize and deserialize JSON for a deep clone